docs/docs/dev_docs/privacy/main.md

```diff
@@ -0,0 +1,165 @@
+- Passing arguments to a public function from a private function. All of those arguments will be publicly visible.
+- Calling an internal public function from a private function. The fact that the call originated from a private function of that same contract will be trivially known.
+- Emitting unencrypted events from a private function. The unencrypted event name and arguments will be publicly visible.
+- Sending L2->L1 messages from a private function. The entire message, and the resulting L1 function execution will all be publicly visible.
+
+
+### Crossing the public -> private boundary
+
+If a public function sends a message to be consumed by a private function, the act of consuming that message might be leaked without following recommended patterns. See (LINK) for more details.
+
+
+### Timing of transactions
+
+Information about the nature of a transaction can be leaked based on the timing of that transaction.
+
+If a transaction is executed at 8am GMT, it's much less likely to have been made by someone in the USA. That's information, right there.
+
+If there's a spike in transactions on the last day of every month, those might be salaries.
+
+Suppose that every time Alice sends Bob a private token, 1 minute later a transaction is always submitted to the tx pool with the same kind of 'fingerprint'. Alice might deduce that these transactions are automated reactions by Bob. (Here, 'fingerprint' is an intentionally vague term. It could be a public function call, or a private tx proof with a particular number of nonzero public inputs, or some other discernible pattern that Alice sees).
+
+Tl;dr: app developers should think about the _timing_ of user transactions, and how this might leak information.
+
+
+### Function Fingerprints and Tx Fingerprints
+
+'Fingerprint' is something of a neologism, in the context of privacy leakage.
+
+A 'Function Fingerprint' is any data which is exposed by a function to the outside world. A 'Tx Fingerprint' is any data which is exposed by a tx to the outside world. We're interested in minimising leakages of information from private txs. For a private function, the following information _could_ be leaked (depending on the function, of course):
+
+- All calls to public functions.
+  - The contract address of the private function (if it calls an internal public function).
+    - This could be the address of the transactor themselves, if the calling contract is an account contract.
+  - All arguments which are passed to public functions.
+- All calls to L1 functions (in the form of L2 -> L1 messages).
+  - The contents of L2 -> L1 messages.
+- All unencrypted logs (topics and arguments).
+- The roots of all trees which have been read from.
+- The _number_ of ['side effects'](https://en.wikipedia.org/wiki/Side_effect_(computer_science)):
+  - \# new commitments
+  - \# new nullifiers
+  - \# bytes of encrypted logs
+  - \# public function calls
+  - \# L2->L1 messages
+  - \# nonzero roots**
```

Comment:
The footnote don't seem to work as planned. But the duplicate seems to be there.